### PR TITLE
Add custom components to mdx integration guide

### PR DESCRIPTION
**Pull request by kylebutts**
## Changes

Following https://github.com/withastro/docs/issues/1234, I've added a section to the integration guide on custom components in MDX

## Docs

This should be automatically pulled into the docs according to: https://github.com/withastro/docs/pull/1426

### CODE REVIEW

File: packages/integrations/mdx/README.md [hunk 1]
```diff
@@ -249,6 +249,44 @@ const { title, fancyJsHelper } = Astro.props;
 <!-- -->
 ```
 
+### Custom components
+
+Under the hood, MDX will convert markdown into html components. For example,
```
Comment (bholmesdev) [inline packages/integrations/mdx/README.md:254]:
```suggestion
Under the hood, MDX will convert markdown into html components. For example, this blockquote:
```

Small nit for clarity!

File: packages/integrations/mdx/README.md [hunk 2]
```diff
@@ -249,6 +249,44 @@ const { title, fancyJsHelper } = Astro.props;
...
 
+### Custom components
+
+Under the hood, MDX will convert markdown into html components. For example, 
+
+```md
+> A blockquote with *some* emphasis.
+```
+
+will be converted into
```
Comment (bholmesdev) [inline packages/integrations/mdx/README.md:260]:
```suggestion
will be converted into this HTML:
```

Nit: we usually end lines with colons for snippets.

File: packages/integrations/mdx/README.md [hunk 3]
```diff
@@ -249,6 +249,44 @@ const { title, fancyJsHelper } = Astro.props;
...
+
+```html
+<blockquote>
+  <p>A blockquote with <em>some</em> emphasis.</p>
+</blockquote>
+```
+
+MDX provides a way to tap into this rendering and use your own custom components. In the above example, you could create a custom Blockquote component (in any language) that has either a `<slot />`  component or accepts a `children` prop. Then in the MDX file you import the component and export it to the `components` export.
```
Comment (bholmesdev) [inline packages/integrations/mdx/README.md:268]:
```suggestion
But what if you want to map these blockquotes to our own markup? In the above example, you could create a custom `Blockquote` component (in any language) that has either a `<slot />`  component or accepts a `children` prop. 

[example here?]

Then, you can map the `blockquote` element to your `Blockquote` component using the `components` export:
```

Nit on explaining the "why" before diving in. Also, I think an example `Blockquote.jsx` component snippet would clarify what "slot" and "children" would be used for!

File: packages/integrations/mdx/README.md [hunk 4]
```diff
@@ -249,6 +249,44 @@ const { title, fancyJsHelper } = Astro.props;
...
+```
+
+MDX provides a way to tap into this rendering and use your own custom components. In the above example, you could create a custom Blockquote component (in any language) that has either a `<slot />`  component or accepts a `children` prop. Then in the MDX file you import the component and export it to the `components` export. 
+
+```mdx
+import Blockquote from '../components/Blockquote.astro';
+export const components = { blockquote: Blockquote };
+```
+
+The advantage of this is it allows for the simplicity of writing in markdown without having to write the custom component or writing a remark/rehype plugin. A full list of components that can have custom components is on the [MDX website](https://mdxjs.com/table-of-components/).
```
Comment (bholmesdev) [inline packages/integrations/mdx/README.md:275]:
```suggestion
The advantage of this is it allows for the simplicity of writing in markdown without having to write the custom component or writing a remark/rehype plugin. Visit the [MDX website](https://mdxjs.com/table-of-components/) for a full list of HTML elements that support custom components.
```

Nit: components -> HTML elements for clarity

File: packages/integrations/mdx/README.md [hunk 5]
```diff
@@ -249,6 +249,44 @@ const { title, fancyJsHelper } = Astro.props;
...
+
+#### Custom components with imported `mdx`
+
+Custom components can also be passed to the components prop when rending imported MDX content.
```
Comment (bholmesdev) [inline packages/integrations/mdx/README.md:279]:
```suggestion
When rendering imported MDX content, custom components can also be passed via the `components` prop:
```

Nit on colons and `code blocks`

**Review by bholmesdev — COMMENTED** [review]
Thanks for the thorough issue and PR! Have some comments on wondering and examples (most of them minor), but I'm open to feedback on any of these edits 👍

**Review by sarah11918 — COMMENTED** [review]
I think this is a FABULOUS addition to the docs, @kylebutts!  🥳 

I've left some suggestions for you and @bholmesdev to consider, mostly to integrate with existing Docs conventions. Looking forward to seeing this included! 🚀

**Comment by kylebutts** [comment]
@sarah11918 @bholmesdev I incorporated all your feedback and left open the two conversations. They are incorporated in my commit, but left the convos for you to look over as you reference

File: packages/integrations/mdx/README.md [hunk 6]
```diff
@@ -249,6 +249,54 @@ const { title, fancyJsHelper } = Astro.props;
...
+
+will be converted into this HTML:
+
+```html
+<blockquote>
+  <p>A blockquote with <em>some</em> emphasis.</p>
+</blockquote>
+```
+
+But what if you want to specify your own markup for these blockquotes? In the above example, you could create a custom `<Blockquote />` component (in any language) that has either a `<slot />`  component or accepts a `children` prop. MDX provides a way to tap into this rendering and use your own custom components. In the above example, you could create a custom Blockquote component (in any language) that has either a `<slot />`  component or accepts a `children` prop.
```
Comment (sarah11918) [inline packages/integrations/mdx/README.md:268]:
```suggestion
But what if you want to specify your own markup for `<blockquote>` elements? In the above example, you could create a custom `<Blockquote />` component (in any language) that either has a `<slot />` component or accepts a `children` prop.
```
Thanks for incorporating the comments @kylebutts! It looks like there's some copy here twice? So I'll clean this up and then happy for you and @bholmesdev to edit for nuance/reasonableness.

Note, making some editorial decisions based strictly on formatting here, so not meant to disapprove of any previous language. I just don't like how it looks to write plural tags and stuff.

**Comment by sarah11918** [comment]
Made a quick commit to remove duplicate text, but unresolved the convo in case either of you still want to edit it! I think I'll be happy with whatever y'all go with from here. Thanks for making some little changes for fitting in with existing Docs!

**Review by bholmesdev — COMMENTED** [review]
Just need a `patch` changeset and we're good to go! Yes, a little overkill for README changes, but it should keep our package releases in-check. [Instructions on adding changesets here](https://github.com/withastro/astro/blob/main/CONTRIBUTING.md#making-a-pull-request)

**Comment by kylebutts** [comment]
All set!

**Review by bholmesdev — APPROVED** [review]
Done! Ready to merge 🚀

**Pull request merged**